### PR TITLE
updated land tiles to use NOAA coastlines and Copernicus S2 kml

### PR DESCRIPTION
**Pull request by freitagb**

### CODE REVIEW

File: README.TXT [hunk 1]
```diff
@@ -1,35 +1,27 @@
...
 How was the list of HLS tiles over land excluding Antarctica derived.
 
-1. Land mask shapefile from
-   https://www.naturalearthdata.com/downloads/10m-physical-vectors/10m-land/
+1. The land mask was derived using the shorelines dataset from NOAA
+   https://www.ngdc.noaa.gov/mgg/shorelines/data/gshhg/latest/
+   1) Shapefiles are not included in the github repo and should be downloaded from the link above.
+   2) The full resolution GSHHG level 1 shapefile is used (GSHHS_shp/f/GSHHS_f_L1.shp)
+	a) Level 1 shapefile includes the land/ocean boundary - inland water bodies are not masked.
+	b) Lower resolution shapefiles are available. If users want to increase processing time with a lower resolution shapefile, update the file path in the params.json file
```
Comment (ciaransweet) [inline README.TXT]: Forgive my naivety - do you mean `reduce processing time with a lower resolution shapefile`?

File: README.TXT [hunk 2]
```diff
@@ -1,35 +1,27 @@
...
+	c) A 0.01 degree buffer is added to the land boundary prior to finding the S2 grid intersection.
 
-   Saved in ne_10m_land/
+2. The KML of the MGRS grid is provided via Copernicus and can be converted as stored as a geojson using create_S2_geojson.py:
```
Comment (ciaransweet) [inline README.TXT]:
```suggestion
2. The KML of the MGRS grid is provided via Copernicus and can be converted and stored as a geojson using create_S2_geojson.py:
```

File: HLS_land_tiles.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import datetime
+import fiona
+import json
+import requests
+import xmltodict
+
+from shapely.geometry import Polygon
+
+
+class hls_land_tiles:
```
Comment (ciaransweet) [inline HLS_land_tiles.py:10]: This should really be in PascalCase

```suggestion
class HLSLandTiles:
```

File: HLS_land_tiles.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        self.get_land_mask(land_shpfile)
+        self.get_S2grid_from_kml(kml_url)
+        self.write_s2_hls_grid()
+
+    def get_land_mask(self,land_shpfile):
+        shape = fiona.open(land_shpfile)
+        for tile in shape:
+            self.land_polys.append(Polygon(tile["geometry"]["coordinates"][0]).buffer(0.01,cap_style=3,join_style=3))
+
+    def get_S2grid_from_kml(self,kml_url):
```
Comment (ciaransweet) [inline HLS_land_tiles.py:27]:
```suggestion
    def get_s2_grid_from_kml(self,kml_url):
```

File: HLS_land_tiles.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        with open("params.json") as f:
+            params = json.load(f)
+        land_shpfile = params["path_to_gshhs_shp"]
+        kml_url = params["S2_kml_url"]
+        self.land_polys = []
+        self.get_land_mask(land_shpfile)
+        self.get_S2grid_from_kml(kml_url)
+        self.write_s2_hls_grid()
+
+    def get_land_mask(self,land_shpfile):
```
Comment (ciaransweet) [inline HLS_land_tiles.py:22]:
```suggestion
    def get_land_mask(self, land_shpfile):
```

File: HLS_land_tiles.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        shape = fiona.open(land_shpfile)
+        for tile in shape:
+            self.land_polys.append(Polygon(tile["geometry"]["coordinates"][0]).buffer(0.01,cap_style=3,join_style=3))
```
Comment (ciaransweet) [inline HLS_land_tiles.py:25]:
```suggestion
            self.land_polys.append(Polygon(tile["geometry"]["coordinates"][0]).buffer(0.01, cap_style=3, join_style=3))
```

File: HLS_land_tiles.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+    def get_S2grid_from_kml(self,kml_url):
+        S2_complete_grid = xmltodict.parse(requests.get(kml_url).text)
```
Comment (ciaransweet) [inline HLS_land_tiles.py:28]:
```suggestion
        s2_complete_grid = xmltodict.parse(requests.get(kml_url).text)
```

File: HLS_land_tiles.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        objects = S2_complete_grid["kml"]["Document"]["Folder"][0]["Placemark"]
```
Comment (ciaransweet) [inline HLS_land_tiles.py:29]:
```suggestion
        objects = s2_complete_grid["kml"]["Document"]["Folder"][0]["Placemark"]
```

File: README.TXT (hunk 2 above)
Comment (ciaransweet) [reply README.TXT]: @freitagb why do you create the GeoJSON? Is it for other purposes than making the HLS list?

**Review by ciaransweet — CHANGES_REQUESTED** [review]
Most of these comments are questions/small vanity changes @freitagb so feel free to ignore/challenge them!

If it makes the right data, then I'd say the codes gucci, seems to do what it should!

**Comment by freitagb** [comment]
Hello,
    
           I recently transitioned to NASA and no longer check my uah
    email. Please reach out to me at my new email address
    ***@***.***
    
    Thanks,
          Brian

**Pull request merged**